File: src/main/java/com/nobility/downloader/entities/settings/Meta.java
package com.nobility.downloader.entities.settings;

public interface Meta {

    String getKey();

    void setKey(String key);

    void setValueObj(Object value);

    Object getValueObj();

    void setStringValue(String stringValue);

    String getStringValue();

    void setIntValue(Integer intValue);

    Integer getIntValue();

    void setLongValue(Long longValue);

    Long getLongValue();

    void setDoubleValue(Double doubleValue);

    Double getDoubleValue();

    void setFloatValue(Float floatValue);

    Float getFloatValue();

    void setBooleanValue(Boolean booleanValue);

    Boolean getBooleanValue();

}

File: src/main/java/com/nobility/downloader/entities/settings/Meta.kt
package com.nobility.downloader.entities.settings

interface Meta {
    var key: String?
    var valueObj: Any?
    var stringValue: String?
    var intValue: Int?
    var longValue: Long?
    var doubleValue: Double?
    var floatValue: Float?
    var booleanValue: Boolean?
}